pkg/engine/renderer/enhancer_test.go

```diff
@@ -140,6 +138,107 @@ func TestEnhancerApply_noAdditionalMetadata(t *testing.T) {
 		assert.False(t, ok, "Pod struct contains template field")
 	}
 }
+func TestEnhancerApply_dependencyHash_noDependencies(t *testing.T) {
+	ss := statefulSet("statefulset", "default")
+
+	tpls := map[string]string{
+		"statefulset": resourceAsString(ss),
+	}
+
+	meta := metadata()
+	meta.PlanUID = uuid.NewUUID()
+
+	e := &DefaultEnhancer{
+		Scheme:    utils.Scheme(),
+		Discovery: fake.CachedDiscoveryClient(),
+	}
+
+	objs, err := e.Apply(tpls, meta)
+	if err != nil {
+		t.Errorf("failed to apply template %s", err)
+	}
+
+	ssApplied := funk.Find(objs, func(o runtime.Object) bool {
+		return o.GetObjectKind().GroupVersionKind() == schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "StatefulSet"}
+	})
+
+	unstructMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(ssApplied)
+	assert.Nil(t, err, "failed to parse object to unstructured: %s", err)
+
+	annotations, _, _ := unstructured.NestedMap(unstructMap, "spec", "template", "metadata", "annotations")
+	assert.NotNil(t, annotations, "Statefulset pod template spec contains no annotations")
+
+	hash := annotations[kudo.DependenciesHashAnnotation]
+	assert.Nil(t, hash, "Pod template spec annotations contains a dependency hash but no dependencies")
+}
+
+func TestEnhancerApply_dependencyHash_changes(t *testing.T) {
```

Comment:
suggestion: for long tests like this, which go through multiple states [ginkgo](https://onsi.github.io/ginkgo/) might offer a better structure (and closer to scalatest). Take a look at the [instance_admission_integration_test.go](https://github.com/kudobuilder/kudo/blob/master/pkg/webhook/instance_admission_integration_test.go#L144)